infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -500,79 +598,26 @@ protected void initChannel(@Nonnull Channel ch) throws Exception {
     /**
      * Cleanly shuts down the server and restarts.
      *
-     * @param serverContext Server Context.
      * @param resetData     Resets and clears all data if True.
      */
-    public static void restartServer(ServerContext serverContext, boolean resetData) {
-
-        final Thread previousServerThread = corfuServerThread;
-        final Map<String, Object> opts = serverContext.getServerConfig();
-        final boolean bindToAllInterfaces = serverContext.isBindToAllInterfaces();
-
-        corfuServerThread = new Thread(() -> {
-            cleanShutdown(serverContext.getServers());
-            if (resetData && !(Boolean) serverContext.getServerConfig().get("--memory")) {
-                File serviceDir = new File((String) serverContext.getServerConfig()
-                        .get("--log-path"));
-                try {
-                    FileUtils.cleanDirectory(serviceDir);
-                } catch (IOException ioe) {
-                    throw new UnrecoverableCorfuError(ioe);
-                }
-            }
-            serverContext.close();
+    static void restartServer(boolean resetData) {
 
-            // Wait for previous server thread to join.
-            try {
-                previousServerThread.join();
-                Runtime.getRuntime().removeShutdownHook(shutdownThread);
-            } catch (InterruptedException ie) {
-                throw new UnrecoverableCorfuInterruptedError(ie);
-            }
-
-            Runtime.getRuntime().removeShutdownHook(shutdownThread);
+        if (resetData) {
+            CLEANUP_SERVER = true;
+        }
 
-            // Restart the server.
-            log.info("RestartServer: Restarting corfu server");
-            printStartupMsg(opts);
-            startServer(opts, bindToAllInterfaces);
-        });
-        corfuServerThread.setName("CorfuServer");
-        corfuServerThread.start();
+        log.info("RestartServer: Shutting down corfu server");
+        ACTIVE_SERVER.close();
+        log.info("RestartServer: Starting corfu server");
```

Comment:
Which lines of code start the corfu server?

Reply:
The server waits on a future to be closed. The restart mechanism just completes this future and the restart is taken care by the while loop.